table-api/src/main/java/io/deephaven/api/Expression.java

```diff
@@ -3,9 +3,10 @@
 public interface Expression {
 
     static Expression parse(String x) {
-        if (ColumnName.isValidColumnName(x)) {
-            return ColumnName.of(x);
-        }
+        // In this context, we can't know if the RHS is a column or not.
```

Comment:
Should we distinguish `Identifier` from raw? Then it could be a query scope var *or* a `ColumnName`.

Reply:
Initially, I liked this idea. Thinking about it more, I'd prefer to not have this semi-parsed statement as part of the syntax tree. During manual construction (ie, construction not from a string) there is no reason why somebody would want to use `Identifier` instead of being more specific, so I don't think it's an appropriate abstraction for this level.

Reply:
I think the key here is to allow us to distinguish between formulas that *might* resolve to column assignments and ones that definitely require compilation.